internal/benchmarks/go.mod

```diff
@@ -0,0 +1,21 @@
+module github.com/jhump/protocompile/internal/benchmarks
```

Comment:
I made this a separate module so that `go test ./...` in the repo root would exclude these.

Also, I didn't want to pull in `github.com/jhump/protoreflect` as a dependency in the root `go.mod`.